crates/cadenza-eval/STATUS.md

```diff
@@ -684,6 +684,98 @@ The following enhancements were identified during code review and are planned fo
 - [ ] **Native targets**: Emit Rust code (primary), Cranelift/LLVM (optional)
 - [ ] **Dead code elimination**: Remove unused specializations
 
+#### WASM Code Generation
+
+**Status**: Initial implementation merged with basic WAT output generation. Critical next step is proper value management.
+
+**Current State**:
+- [x] Basic WASM codegen infrastructure with wasm-encoder
+- [x] WAT snapshot testing for all test-data files
+- [x] Function type signatures and declarations
+- [x] Simple constant generation (integers, floats, booleans)
+- [x] Binary operations (arithmetic, comparison) - instruction generation only
+- [x] Module structure with proper section ordering
+- [x] ~~Value location tracking with local management~~ ✅ (Completed - all 89 WAT tests pass)
+
+**Critical Issues** (blocking correct WASM output):
+1. ~~**Value Location Tracking**~~: ✅ **FIXED** - WASM codegen now properly tracks SSA values as WASM locals and generates correct `local.get`/`local.set` instructions.
+
+**High Priority Tasks** (in order):
+- [x] ~~**Value location tracking and local management**~~  ✅
```

Comment:
Longer term we should really track how a const value is derived. It's definitely an interesting problem with having evaluation done at compile time, when you get an IR const value it may or may not have a syntax node associated with it. It could be a computed value. And if it is computed how do you know where it came from. Or is it even worth tracking it? What do we get from it?